### DISCUSSION TITLE
How to type the data that goes through a middleware

### DISCUSSION THREAD

**Discussion by Satyam** [vercel/swr#1677] [Q&A]
In `useSWR` I can declare the type of the data returned with `useSWR<Sale[], Error>`..  

However, I couldn't find a way to tell the middleware I configured via `{ use: [salesMiddleware] }` which I am using to transform part of the data fetched.  

I just declared it as:

```js
const salesMiddleware: Middleware = (useSWRNext) => {
```
but I couldn't find any way to override the declaration for the data.  After several attempts at typecasting the data inside the middleware function, which looks awful, I ended up with `// @ts-ignore`, which is not quite the proper thing to do but, by the time being, it looks less awful than all the messy type casting here and there.

BTW: it might be nice to add typings to the [examples for the middleware](https://swr.vercel.app/docs/middleware#examples) .

Thanks in advance for the help.

**Comment by mrfrodo24** [comment]
Bump.  As far as I can tell this is still unsupported.

**Comment by patjakubik** [comment]
3 years later, also wondering about this!
